Plugin support to add items to the homepage (#674)
nautobot/nautobot: nautobot/docs/plugins/development.md

* Inject template content and navigation links
* Establish their own GraphQL types and REST API endpoints
* Add custom request/response middleware

> * Create top level navigation items
> * Modify existing navigation items
> * Create new panels for homepage

Reviewer: ```suggestion
* Add content to the Nautobot navigation menu bar
* Add content to the Nautobot home page
```

    - datasources.py        # Loading Data from a Git Repository
    - graphql/
      - types.py            # GraphQL Type Objects

>     - homepage.py           # Homepage Panel

Reviewer: ```suggestion
    - homepage.py           # Home Page Content
```

More documentation and examples can be found [here](../development/navigation-menu.md)

> ## Homepage Panels
> 
> Plugins can create panels in the homepage by defining `layout` inside of `homepage.py`. Using the key and weight system, a developer can create new panels using the pre-existing items or by using custom templates with callback functions to collect data.

Reviewer: ```suggestion
## Home Page Content

Plugins can add content to the Nautobot home page by defining `layout` inside of `homepage.py`. Using the key and weight system, a developer can integrate the plugin content amongst existing panels, groups, and items and/or create entirely new panels as desired.
```
